### PR TITLE
test(ods): raise ods test coverage to 90%

### PR DESCRIPTION
**Pull request by jmelahman**
## Issues to Address

`tools/ods` had 39.1% statement coverage. Many commands had no tests, because they call external tools or exit the process with `log.Fatal`. This PR adds tests for almost the whole module and raises the floors in `.coverage-baseline.yaml`.

Total coverage goes from **39.1% to 90.5%**.

| Package | Floor before | Floor after |
| --- | --- | --- |
| `cmd` | 23.3 | 90.4 |
| `internal/alembic` | 0 | 98.7 |
| `internal/audit` | 52.1 | 98 |
| `internal/auditcmd` | 0 | 85.2 |
| `internal/basedigest` | 54 | 97.5 |
| `internal/childproc` | 0 | 100 |
| `internal/composegen` | 97.3 | 99.3 |
| `internal/config` | 0 | 94.7 |
| `internal/coverage` | 92.3 | 97.1 |
| `internal/deployfilessync` | 79.1 | 91.6 |
| `internal/docker` | 23.4 | 99.2 |
| `internal/git` | 42.9 | 94.7 |
| `internal/imgdiff` | 75.5 | 97.5 |
| `internal/kube` | 0 | 100 |
| `internal/lazyimports` | 62.8 | 95.5 |
| `internal/openapi` | 0 | 93.9 |
| `internal/paths` | 30.3 | 74.2 |
| `internal/portutil` | 76.9 | 96.1 |
| `internal/postgres` | 0 | 100 |
| `internal/prompt` | 0 | 62.1 |
| `internal/pycheck` | 74.7 | 94 |
| `internal/release` | 85.8 | 95.6 |
| `internal/s3` | 21.1 | 98 |
| `internal/terraform` | 82.3 | 97 |
| `internal/testsuite` | 88.3 | 97 |
| `internal/tui` | 41.9 | 94.6 |
| `internal/typecoverage` | 95.4 | 100 |

## Important Notes

**How the tests isolate external tools**
- The tests put fake `#!/bin/sh` scripts for `docker`, `gh`, `aws`, `kubectl`, `uv`, `bun`, `lsof` and other tools on `PATH`. The scripts record their argv, and the tests check the exact arguments. These tests skip on Windows.
- Git tests use temporary repositories from `internal/gittest`. HTTP goes to `httptest` servers, and registry lookups go to an in-process registry.
- Each test uses `t.TempDir`, `t.Setenv` and `t.Chdir`. No test uses the network, a docker daemon, a database or the developer's home config.
- The suite passes with `-race -shuffle=on`.

**Production changes (behaviour is unchanged)**
- **Run bodies:** Run closures that called `log.Fatal` now call `runX(...) error` functions. Run logs the returned error with `log.Fatal`, so messages and exit codes do not change.
- **`fatalErrorf`:** a new helper in `cmd` builds these errors. It keeps the capitalized log text and passes the ST1005 lint check.
- **Test inputs as parameters:** some functions now take their external inputs as plain parameters:
  - An `io.Writer` or `io.Reader` in `trace`, `whois`, `check-getattr`, `lint` and `fmt`.
  - The OSV query URL in `internal/audit`.
  - The polling intervals in the deploy watchers.
  - A `tcell.Screen` in the TUI picker and select loops.
  - A reader in the `internal/prompt` loops.
- **Temp directory cleanup:** `screenshot-diff` now deletes its temporary download directories when it fails. Before, `log.Fatal` skipped the deferred cleanup.
- **`portutil` floor:** the `portutil` tests fake `lsof` and `ps`. The floor no longer depends on whether the host has `lsof`.

**Not covered**
- The `log.Fatal` and `os.Exit` lines in Run closures, and `main.go` and `odsaudit/main.go`.
- Interactive confirmation prompts. `internal/prompt` reads a stdin reader created at package init, so the tests use `--yes` or similar flags.
- Code that needs a real terminal (`tcell.NewScreen`), or that always tries real S3 first (`s3.FetchToFile`).
- File system errors that cannot occur when tests run as root, Windows-only branches, and unreachable branches.

**Found while testing (not fixed in this PR)**
- `cmd/backend.go`: the EE default values are added after the `.env` values, so `ENABLE_PAID_ENTERPRISE_EDITION_FEATURES=false` in `.env` is overridden. The comment says `.env` takes precedence.
- `cmd/desktop.go`: `nodeModulesNeedsInstall` checks `node_modules/node_modules`, so `bun install` runs every time.
- The release and deploy tests fail when the global git config sets `tag.gpgsign = true`. `commit.gpgsign` does not affect them.

## Tests

- `go test -race -count=2 -shuffle=on ./...` passes.
- `go vet ./...` and `golangci-lint run ./...` (v2.13.2) report no issues.
- `ods coverage ods --update` generated the new floors, and `ods coverage ods --check` passes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by jmelahman** [comment]
@greptile review

**Comment by jmelahman** [comment]
@greptile review

**Comment by jmelahman** [comment]
@greptile review

File: tools/ods/internal/testsuite/testsuite_test.go [hunk 1]
```diff
@@ -247,3 +279,13 @@ func TestHasTarget(t *testing.T) {
 		})
 	}
 }
+
+// All hands out a copy, so a caller cannot rewrite the routing table.
+func TestAll_returnsACopy(t *testing.T) {
+	all := All()
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/testsuite/testsuite_test.go:286]: No change: `All` deep-copies `Aliases` and `DefaultArgs` with `slices.Clone`, and `TestAll_returnsACopy` already mutates every nested element and checks the table stays the same.

File: tools/ods/internal/audit/image_run_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+func TestRunImage_imageWithoutPackagesIsClean(t *testing.T) {
+	bin := fakeBinDir(t)
+	tmp := t.TempDir()
+	t.Setenv("TMPDIR", tmp)
+	dockerArgs := fakeDockerWithImage(t, bin)
+	allowlist := writeFixture(t, t.TempDir(), "ignores.json", `{"ignores":[]}`)
+
+	var stdout, stderr bytes.Buffer
+	res, err := RunImage(ImageOptions{
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/audit/image_run_test.go:56]: No change: `scanImage` does not set `CompareOffline`, so osv-scanner downloads no database. The fixture image has no packages, so the scan returns `ErrNoPackagesFound` before any query. The tests pass with HTTP(S)_PROXY pointed at a closed port.

File: tools/ods/internal/kube/kube_test.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+	"testing"
+)
+
+// fakeTools puts shell scripts named after each key on PATH, and nothing else.
+// Every call appends its argv (tool name first) to a log that calls reads back.
+func fakeTools(t *testing.T, scripts map[string]string) func() [][]string {
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/kube/kube_test.go:14]: No change: `cmd` and `internal/kube` are different packages. Sharing a _test.go helper would need a new non-test package just for this fake, which costs more than the small duplicate.

File: tools/ods/internal/git/commands_test.go [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+	repo.Git("checkout", "--quiet", "-b", "topic")
+	topic := repo.Commit("topic change", "README.md", "topic")
+	repo.Git("checkout", "--quiet", "main")
+	repo.Commit("main change", "README.md", "main")
+	repo.Commit("main extra", "extra.txt", "extra")
+
+	if HasMergeConflict() || IsCherryPickInProgress() || HasStagedChanges() || IsRebaseInProgress() {
+		t.Fatal("expected no conflict state before the cherry-pick")
+	}
+	if n, err := CountUniqueCommits("main", "topic"); err != nil || n != 2 {
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/git/commands_test.go:480]: No change: `CountUniqueCommits(branch, upstream)` runs `git rev-list --count upstream..branch`, so `("main", "topic")` counts the 2 commits on main that are not on topic. That matches the comment and the assertion, and the test passes.

File: tools/ods/internal/auditcmd/fakes_test.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	dir := t.TempDir()
+	t.Setenv("PATH", dir+string(os.PathListSeparator)+os.Getenv("PATH"))
+	return dir
+}
+
+// writeFakeCommand writes an executable shell script named name into dir. Each
+// call appends its arguments as one line to <dir>/<name>.args before running
+// body, and the returned function reads those lines back.
+func writeFakeCommand(t *testing.T, dir, name, body string) func() []string {
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/auditcmd/fakes_test.go:29]: No change: `_test.go` helpers cannot be shared across packages without a new non-test package. The two helpers now record calls in different formats for their own assertions.

File: tools/ods/cmd/lint_test.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	file := filepath.Join(dir, "main.tf")
+	writeFile(t, file, "a = \"123456789012\"\n")
+
+	var stderr bytes.Buffer
+	clean, err := runLintTerraform([]string{file}, &stderr)
+	if err != nil {
+		t.Fatalf("runLintTerraform: %v", err)
+	}
+	// With no repository root the path is shown as given.
+	if clean || !strings.Contains(stderr.String(), "  "+file+":1: ") {
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/cmd/lint_test.go:71]: No change: outside a repository `GitRoot` returns `""`, and `filepath.Rel("", <absolute path>)` returns an error. So `relativeTo` keeps the path as given, which the test expects. The test passes.

File: tools/ods/internal/lazyimports/lazyimports_repo_test.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+	"path/filepath"
+	"slices"
+	"testing"
+
+	"github.com/onyx-dot-app/onyx/tools/ods/internal/gittest"
+)
+
+// newBackendRepo creates a git repository, writes files under its backend
+// directory, and makes the root the working directory.
+func newBackendRepo(t *testing.T, files map[string]string) string {
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/lazyimports/lazyimports_repo_test.go:15]: No change: this short fixture is in a different package from `internal/pycheck`. Sharing it would need a new non-test package only for tests.

**Comment by jmelahman** [comment]
@greptile review

File: tools/ods/internal/s3/fetch.go [hunk 8]
```diff
@@ -155,10 +161,18 @@ func fetch(s3url string, destPath string, quiet bool) error {
 	return fmt.Errorf("failed to download from S3: %w\n\nTo authenticate, run:\n  aws sso login\n\nOr configure AWS credentials with:\n  aws configure sso", cliErr)
 }
 
+// httpClient bounds the wait for a response. It sets no overall timeout,
+// because large objects can take long to download.
+var httpClient = func() *http.Client {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.ResponseHeaderTimeout = 30 * time.Second
+	return &http.Client{Transport: transport}
+}()
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply tools/ods/internal/s3/fetch.go]: Fixed: `newHTTPClient` now checks the assertion. It uses a custom round tripper as it is, and adds the header timeout only to an `*http.Transport` copy. A test covers both cases.

**Comment by jmelahman** [comment]
@greptile review

**Review by justin-tahara — APPROVED** [review]
